website/blog/2024-08-13-distributed-execution-flow-paradigms/index.mdx

```diff
@@ -0,0 +1,335 @@
+
+- [Temporal](https://temporal.io/)
+- [Azure Durable Functions](https://learn.microsoft.com/en-us/azure/azure-functions/durable/)
+- [Amazon SWF](https://docs.aws.amazon.com/swf/)
+- [Uber Cadence](https://cadenceworkflow.io/)
+- [Infinitic](https://www.infinitic.io/)
+
+Among these, [Temporal](https://temporal.io/) has grown in influence, used by companies like SnapChat, HashiCorp, Stripe, DoorDash, and DataDog. Its success is driven by its practical application in real-world scenarios and the expertise of its founders.
+
+At Metatype, we recognize the value of durable execution and are committed to making it accessible. Our [Temporal Runtime](/docs/reference/runtimes/temporal) integrates seamlessly into our declarative API development platform, enabling users to harness the power of Temporal directly within Metatype. For those interested in exploring further, our documentation provides a detailed guide on getting started with [Temporal Runtime](/docs/reference/runtimes/temporal).
+
+Below is an example of how you can build a simple API to interact with an order delivery temporal workflow within Metatype.
+
+:::note
+If you are new to Metatype or haven’t set it up yet in your development environment. You can follow this [guideline](/docs/tutorials/quick-start).
+:::
+
+For this example, the order delivery system will have few components/services such as Payment, Inventory and Delivery.
+
+Your temporal workflow definition should look similar to the one below.
+
+<SDKTabs>
+
+    <TabItem value="typescript">
+
+```typescript
+import { Worker, WorkflowClient } from "@temporalio/client";
+import { proxyActivities } from "@temporalio/workflow";
+
+export const { processPayment, checkInventory, deliverOrder } =
+  proxyActivities<{
+    processPayment(orderId: string): Promise<string>;
+    checkInventory(orderId: string): Promise<string>;
+    deliverOrder(orderId: string): Promise<string>;
+  }>({
+    startToCloseTimeout: "10 seconds",
+  });
+
+export async function orderWorkflow(orderId: string): Promise<string> {
+  const paymentResult = await processPayment(orderId);
+  const inventoryResult = await checkInventory(orderId);
+  const deliveryResult = await deliverOrder(orderId);
+  return `Order ${orderId} completed with results: ${paymentResult}, ${inventoryResult}, ${deliveryResult}`;
+}
+
+async function main() {
+  const client = new WorkflowClient();
+  const result = await client.execute(orderWorkflow, {
+    taskQueue: "order-task-queue",
+    workflowId: "order123",
+    args: ["order123"],
+  });
+  console.log(result);
+}
+
+main().catch((err) => {
+  console.error(err);
+  process.exit(1);
+});
+```
+
+Down below is the workflow communication bridge for this system expressed within a [Typegraph](/docs/reference/typegraph).
+
+```typescript
+import { Policy, t, typegraph } from "@typegraph/sdk/index.ts";
+import { TemporalRuntime } from "@typegraph/sdk/providers/temporal.ts";
+
+function getEnvVariable(
```

Comment:
suggestion: You don't really need to show this. You can just use the node approach for the article.